packages/xo-server-sdn-controller/src/index.js

```diff
@@ -715,6 +671,70 @@ class SDNController extends EventEmitter {
     }
   }
 
+  async _addHostToPoolNetworks(host) {
+    const xapi = host.$xapi
+
+    const tunnels = filter(xapi.objects.all, { $type: 'tunnel' })
+    for (let i = 0; i < tunnels.length; ++i) {
+      const tunnel = tunnels[i]
+      const accessPIF = await xapi._getOrWaitObject(tunnel.access_PIF)
+      if (accessPIF.host !== host.$ref) {
+        continue
+      }
+
+      const poolNetwork = find(this._poolNetworks, {
+        network: accessPIF.network,
+      })
+      if (poolNetwork == null) {
+        continue
+      }
+
+      if (accessPIF.currently_attached) {
+        continue
+      }
+
+      log.debug(
+        `Pluging PIF: '${accessPIF.device}' for host: '${
+          host.name_label
+        }' on network: '${accessPIF.$network.name_label}'`
+      )
+      try {
+        await xapi.call('PIF.plug', accessPIF.$ref)
+      } catch (error) {
+        log.error(
+          `XAPI error while pluging PIF: '${accessPIF.device}' on host: '${
+            host.name_label
+          }' for network: '${accessPIF.$network.name_label}'`
+        )
```

Comment:
What if the error is `typeerror cannot read property '$ref' of undefined` then here you're logging something that isn't related to the real error.
2 solutions:
- Log the error.message with rest of information
- Make a specific error with the library `make-error` and re throw it

Reply:
It's wanted to not show this error to the user because it can happen in a normal scenario when trying to plug an already plugged `PIF`.